src/Battlescape/ActionMenuState.cpp

```diff
@@ -180,6 +180,56 @@ ActionMenuState::ActionMenuState(BattleAction *action, int x, int y) : _action(a
 	{
 		addItem(BA_USE, "STR_USE_MIND_PROBE", &id, Options::keyBattleActionItem1);
 	}
+	else if (weapon->getBattleType() == BT_SCRIPTED)
+	{
+		std::vector<SDLKey> hotKeys = {Options::keyBattleActionItem1, Options::keyBattleActionItem2, Options::keyBattleActionItem3, Options::keyBattleActionItem4, Options::keyBattleActionItem5};
+
+		if (!hotKeys.empty() && weapon->getCostUse().Time > 0)
+		{
+			addItem(BA_USE, "STR_USE_SCRIPTED", &id, hotKeys.back());
+			hotKeys.pop_back();
+		}
+		if (!hotKeys.empty() && weapon->getCostSnap().Time > 0)
+		{
+			addItem(BA_SNAPSHOT, "STR_SNAP_SCRIPTED", &id, hotKeys.back());
+			hotKeys.pop_back();
+		}
+		if (!hotKeys.empty() && weapon->getCostPanic().Time > 0)
+		{
+			addItem(BA_PANIC, "STR_PANIC_SCRIPTED", &id, hotKeys.back());
+			hotKeys.pop_back();
+		}
+		if (!hotKeys.empty() && weapon->getCostAimed().Time > 0)
+		{
+			addItem(BA_AIMEDSHOT, "STR_AIMED_SCRIPTED", &id, hotKeys.back());
+			hotKeys.pop_back();
+		}
+		if (!hotKeys.empty() && weapon->getCostMind().Time > 0)
+		{
+			addItem(BA_MINDCONTROL, "STR_MIND_SCRIPTED", &id, hotKeys.back());
+			hotKeys.pop_back();
+		}
+		if (!hotKeys.empty() && weapon->getCostAuto().Time > 0)
+		{
+			addItem(BA_AUTOSHOT, "STR_AUTO_SCRIPTED", &id, hotKeys.back());
+			hotKeys.pop_back();
+		}
+		if (!hotKeys.empty() && weapon->getCostMelee().Time > 0)
+		{
+			addItem(BA_HIT, "STR_MELEE_SCRIPTED", &id, hotKeys.back());
+			hotKeys.pop_back();
+		}
+		if (!hotKeys.empty() && weapon->getCostPrime().Time > 0)
+		{
+			addItem(BA_PRIME, "STR_PRIME_SCRIPTED", &id, hotKeys.back());
+			hotKeys.pop_back();
+		}
+		if (!hotKeys.empty() && weapon->getCostUnprime().Time > 0)
+		{
+			addItem(BA_UNPRIME, "STR_UNPRIME_SCRIPTED", &id, hotKeys.back());
+			hotKeys.pop_back();
+		}
```

Comment:
is BA_LAUNCH missing by design or by accident?

Reply:
Bit of both, BA_LAUNCH was outside the scope.